### PR TITLE
feat: enable monitoring by default and add Grafana dashboard

### PR DESCRIPTION
**Pull request by ckaenzig**
## Description of the changes

Enable monitoring in chart and add a dashboard. Choice of dashboard needs to be discussed.

This PR depends on:

- [x] #40

## Breaking change

- [x] No

## Tests executed on which distribution(s)

- [ ] KinD
- [ ] AKS (Azure)
- [x] EKS (AWS)
- [ ] Scaleway
- [ ] SKS (Exoscale)

### CODE REVIEW

File: locals.tf [hunk 1]
```diff
@@ -1,6 +1,32 @@
...
+      deployment = {
+        replicas = var.replicas
+      }
+      metrics = {
+        prometheus = {
+          service = {
+            enabled = true
+          }
+          serviceMonitor = {
+            enabled = true
```
Comment (lentidas) [inline locals.tf:13]: I propose you add the variable you proposed on the `cert-manager` module.

File: variables.tf [hunk 2]
```diff
@@ -25,6 +25,12 @@ variable "namespace" {
   default = "traefik"
 }
 
+variable "replicas" {
+  description = "Number of traefik pods"
```
Comment (lentidas) [inline variables.tf:29]:
```suggestion
  description = "Number of Traefik pods to be deployed."
```

**Review by lentidas — CHANGES_REQUESTED** [review]

File: locals.tf (hunk 1 above)
Comment (ckaenzig) [reply locals.tf:13]: Yep, good point :+1:

File: variables.tf (hunk 2 above)
Comment (ckaenzig) [reply variables.tf:29]: This PR will be rebased on main after #40 is merges.

**Comment by ckaenzig** [comment]
PR rebased on main. Still need to test the grafana dashboard and review infos by Mohamed.

**Comment by ckaenzig** [comment]
Converting back to draft because the dashboard doesn't work in another cluster than first tested, probably because of hard-coded datasource UIDs. To be investigated.

File: locals.tf [hunk 3]
```diff
@@ -4,8 +4,17 @@ locals {
       deployment = {
         replicas = var.replicas
       }
+      metrics = {
+        prometheus = {
+          service = {
+            enabled = true
+          }
+          serviceMonitor = var.enable_service_monitor ? {
+            foo = "bar"
```
Comment (lentidas) [inline locals.tf:13]: I don't understand the line `foo = bar`. Is it just to have a value set in the `values.yaml` in order to activate the dashboard or am I misunderstanding?

File: variables.tf [hunk 4]
```diff
@@ -35,6 +35,12 @@ variable "replicas" {
   default     = "2"
 }
 
+variable "enable_service_monitor" {
+  description = "Enable prometheus ServiceMonitor in helm chart."
```
Comment (lentidas) [inline variables.tf:39]:
```suggestion
  description = "Enable Prometheus ServiceMonitor in the Helm chart."
```

**Review by lentidas — CHANGES_REQUESTED** [review]

File: locals.tf (hunk 3 above)
Comment (ckaenzig) [reply locals.tf:13]: Yes, `var.enable_service_monitor` needs to evaluate to `true` in the helm chart, but we don't need to set any value in it. I put `enabled = true` initially, but that would make people thing that the `enable` value would be used in the chart.

Comment (lentidas) [reply locals.tf:13]: That's what I thought. I agree that putting `enabled = true` could induce someone into error. Maybe just add a small comment on the side so we do not forget the reason for this?

**Pull request merged**